lib/matplotlib/backend_bases.py

```diff
@@ -2446,20 +2446,51 @@ def button_press_handler(event, canvas, toolbar=None):
 
 
 class NonGuiException(Exception):
+    """Raised when trying show a figure in a non-GUI backend."""
     pass
 
 
 class FigureManagerBase(object):
-    """
-    Helper class for pyplot mode, wraps everything up into a neat bundle
+    r"""
+    A backend-independent abstraction of a figure container and controller.
+
+    The figure manager is used by pyplot to interact with the window in a
+    backend-independent way. It's an adapter for the real (GUI) framework that
+    represents the visual figure on screen.
+
+    GUI backends define from this class to translate common operations such
+    as *show* or *resize* to the GUI-specific code. Non-GUI backends do not
+    support these operations an can just use the base class.
+
+    This following basic operations are accessible:
+
+    **Window operations**
+
+    - `~.FigureManagerBase.show`
+    - `~.FigureManagerBase.destroy`
+    - `~.FigureManagerBase.full_screen_toggle`
+    - `~.FigureManagerBase.resize`
+    - `~.FigureManagerBase.get_window_title`
+    - `~.FigureManagerBase.set_window_title`
+
+    **Key and mouse button press handling**
+
+    The manager allows to process matplotlib `.Event`\s.
```

Comment:
The thing the ` FigureManagerBase` does in relation to key events is to hook up the default `key_press_handler` to the `Canvas` (which is what handles the bridge from the GUI event system to our internal system).

Reply:
Thanks. Is that something a user would in any way interact with (like `set_window_title()`) or is that just a technical aspect that is not relevant to the user. In particular, will the user call `key_press()` under any circumstances?

Reply:
Maybe for testing or in some edge-cases, but I would not expect it in normal course of operation.  

I suspect the reason `key_press` is a method is so that sub-classes could over-ride exactly what it did without having to handle the `mpl_connect` call and bookkeeping (but we don't actually do that internally).